freqtrade/freqtradebot.py

```diff
@@ -1227,24 +1240,41 @@ def execute_trade_exit(
         self.strategy.lock_pair(trade.pair, datetime.now(timezone.utc),
                                 reason='Auto lock')
 
-        self._notify_exit(trade, order_type)
+        self._notify_exit(trade, order_type, sub_trade=bool(sub_trade_amt), order=order)
```

Comment:
There is no reason to use the order dict here.
please use `order_obj` to get type safety and avoid nasty bugs (ccxt/exchange order return values can be "pretty" empty depending on the exchange and endpoint...).

Reply:
Updated